improver/cli/nowcast_optical_flow_perturbation.py

```diff
@@ -0,0 +1,109 @@
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to calculate optical flow advection velocities as perturbations
+from a previous estimate"""
+
+from improver import cli
+
+# Creates the value_converter that clize needs.
+inputadvection = cli.create_constrained_inputcubelist_converter(
+    "precipitation_advection_x_velocity", "precipitation_advection_y_velocity",
+)
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    current_obs: cli.inputcube,
+    previous_forecast: cli.inputcube,
+    previous_advection: inputadvection,
+    orographic_enhancement: cli.inputcube,
+    *,
+    forecast_period=15,
+):
+    """Calculate optical flow components as a perturbation from the previous
+    advection components, using the difference between a current observation
+    and forecast from the previous time step.
+
+    Args:
+        current_obs (iris.cube.Cube):
+            Cube containing latest radar observation
+        previous_forecast (iris.cube.Cube):
+            Cube containing advection nowcast for this time, created at an
+            earlier timestep
+        previous_advection (iris.cube.CubeList):
+            Advection velocities in the x- and y- directions used to generate
+            the previous forecast
+        orographic_enhancement (iris.cube.Cube):
+            Cube containing orographic enhancement for this timestep
+        forecast_period (int):
+            Forecast period of previous forecast, in minutes.  Used to extract
+            a forecast from a cube that may contain multiple lead times.
+
+    Returns:
+        iris.cube.CubeList:
+            List of the umean and vmean cubes.
+
+    """
+    import iris
+    from iris.cube import CubeList
+
+    from improver.nowcasting.optical_flow import OpticalFlow
+    from improver.nowcasting.utilities import ApplyOrographicEnhancement
+
+    # extract forecast at required lead time (expected in seconds)
+    previous_forecast.coord("forecast_period").convert_units("seconds")
+    previous_forecast = previous_forecast.extract(
```

Comment:
Does this do something sensible if it doesn't find a previous forecast?

Reply:
I think that's probably ok . I'm not going to ask you to add an extra error in now.